marketplace-service/src/main/java/com/axonivy/market/service/impl/ImageServiceImpl.java

```diff
@@ -58,6 +64,31 @@ public Image mappingImageFromGHContent(Product product, GHContent ghContent, boo
     return imageRepository.save(image);
   }
 
+  @Override
+  public Image mappingImageFromDownloadedFolder(Product product, Path imagePath) {
+    List<Image> existingImages = imageRepository.findByProductId(product.getId());
+    try {
+      InputStream contentStream = MavenUtils.extractedContentStream(imagePath);
+      byte[] sourceBytes = IOUtils.toByteArray(contentStream);
```

Comment:
IOUtils.toByteArray => can throw NPE if contentStream, but you did not catch NPE